nemo/collections/common/parts/preprocessing/manifest.py

```diff
@@ -75,9 +84,44 @@ def item_iter(
         logging.debug('Using manifest file: %s', str(manifest_file))
         cached_manifest_file = DataStoreObject(manifest_file).get()
         logging.debug('Cached at: %s', str(cached_manifest_file))
+
+        if shard_strategy is not None and world_size > 1:
+            if shard_strategy == 'scatter':
+                # Read only necessary manifest lines
+                logging.debug('Partitioning manifest: global rank = %d, world size = %d', global_rank, world_size)
+
+                # Get number of lines in the current manifest
+                num_lines = get_num_lines(cached_manifest_file)
+                logging.debug('Found %d lines in manifest %s', num_lines, cached_manifest_file)
+
+                # Determine start and end lines in the manifest.
+                # This corresponds to splitting the manifest in audio_to_text.py::expand_audio_filepaths
+                begin_idx = (num_lines // world_size) * global_rank
+                end_idx = begin_idx + num_lines // world_size
+                logging.info(
+                    'Partitioning manifest lines: process (%d) of (%d) taking lines [%d, %d)',
+                    global_rank,
+                    world_size,
+                    begin_idx,
+                    end_idx,
+                )
+            elif shard_strategy == 'replicate':
+                logging.info('Manifest will be replicated across all nodes.')
+                begin_idx = end_idx = None
+            else:
+                raise ValueError(f'Unknown split strategy {shard_strategy}')
+        else:
+            begin_idx = end_idx = None
+
         with open(expanduser(cached_manifest_file), 'r') as f:
-            for line in f:
+            for n, line in enumerate(f):
                 k += 1
+                if begin_idx is not None and n < begin_idx:
+                    # Skip lines before begin_idx
+                    continue
+                if end_idx is not None and n >= end_idx:
+                    # Stop parsing lines from end_idx
+                    break
                 item = parse_func(line, manifest_file)
                 item['id'] = k
```

Comment:
I'm not really sure what this "id" is used for, but if it's important to preserve the original mapping, then you'd need to continue the loop until the end to make sure it's properly counted, I guess? And if it's not important to have it the same, then you can move the `k += 1` to after the ifs, so that it's going from 0 to N on all processes without skips.